test/unit/auth/auth-config.spec.ts

```diff
@@ -296,6 +312,148 @@ describe('MultiFactorAuthConfig', () => {
         }).to.throw(`"${factorId}" is not a valid "AuthFactorType".`);
       });
     });
+
+    it('should build server request with TOTP enabled', () => {
+      expect(MultiFactorAuthConfig.buildServerRequest({
+        state: 'DISABLED',
+        providerConfigs: [
+          {
+            state: 'ENABLED',
+            totpProviderConfig: {},
+          },
+        ],
+      })).to.deep.equal({
+        state: 'DISABLED',
+        providerConfigs: [
+          {
+            state: 'ENABLED',
+            totpProviderConfig: {},
+          },
+        ],
+      });
+    });
+
+    it('should build server request with TOTP disabled', () => {
+      expect(MultiFactorAuthConfig.buildServerRequest({
+        state: 'DISABLED',
+        providerConfigs: [
+          {
+            state: 'DISABLED',
+          },
+        ],
+      })).to.deep.equal({
+        state: 'DISABLED',
+        providerConfigs: [
+          {
+            state: 'DISABLED',
+          },
+        ],
+      });
+    });
+
+    it('should return expected server request on valid TOTP provider config', () => {
+      expect(MultiFactorAuthConfig.buildServerRequest({
+        state: 'DISABLED',
+        providerConfigs: [
+          {
+            state: 'ENABLED',
+            totpProviderConfig: {
+              adjacentIntervals: 5,
+            },
+          },
+        ],
+      })).to.deep.equal({
+        state: 'DISABLED',
+        providerConfigs: [
+          {
+            state: 'ENABLED',
+            totpProviderConfig: {
+              adjacentIntervals: 5,
+            },
+          },
+        ],
+      });
+    });
```

Comment:
Please refactor the buildServerRequest to use one base object, and deepcopy the object in each tests with needed field value updated specifically for the current test.
e.g
```const expectedRequest = { ...};
const expectedDisabledRequest = deepCopy(expectedRequest);
expectedDisabledRequest.state = 'ENABLED';
```
This way it added readability and removed redundant code.